src/ArmTemplates/Common/FileHandlers/FileWriter.cs

```diff
@@ -1,10 +1,37 @@
-﻿using Newtonsoft.Json;
+﻿using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Newtonsoft.Json;
 using System.IO;
+using System.Threading.Tasks;
 
 namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.FileHandlers
 {
     public static class FileWriter
     {
+        public static async Task SaveAsJsonAsync<T>(T item, string fileName)
+        {
+            var serializedItem = item.Serialize();
+            await File.WriteAllTextAsync(fileName, serializedItem);
+        }
+
+        public static async Task SaveAsJsonAsync<T>(T item, string directory, string fileName)
+        {
+            var serializedItem = item.Serialize();
+            await File.WriteAllTextAsync(Path.Combine(directory, fileName), serializedItem);
+        }
+
+        public static async Task SaveAsJsonAsync<T>(T item, params string[] pathsToFile)
+        {
+            var serializedItem = item.Serialize();
+            await File.WriteAllTextAsync(Path.Combine(pathsToFile), serializedItem);
+        }
+
+        public static async Task SaveTextToFileAsync(string xmlContent, params string[] pathsToFile)
+        {
+            await File.WriteAllTextAsync(Path.Combine(pathsToFile), xmlContent);
+        }
+
+        #region Legacy writing methods
```

Comment:
Why do we need to keep them at all?

Reply:
By now didn't get rid of other extractors. The same for the `EntityExtractorBase`.

I just need to make sure they all are used correctly. I am a little bit scared to change it without proper understanding how older methods are called.

Also changing like everything leads to 1000 files PR and I think it will be impossible to review